src/components/Cards/ForecastCard.tsx

```diff
@@ -45,43 +39,71 @@ const ForecastCard = () => {
                 flexDirection={"column"}
                 justifyContent={"space-between"}
             >
-                {weatherForecast.map((day, index) => {
-                    const shouldHaveDivider = index !== weatherForecast.length - 1;
-                    return (
-                        <Fragment key={index}>
+                {forecastData ? (
+                    <ForecastWeather
+                        weatherForecast={days.map((day) => forecastData[`${day}Day`])}
+                    />
+                ) : (
+                    <LoadingScreen />
+                )}
+            </Box>
+        </Box>
+    );
+};
+
+const LoadingScreen = () => {
+    return (
+        <Box
+            sx={{ width: "100%", height: "100%", display: "flex", flexDirection: "column", pb: 25 }}
+        >
+            <Box sx={{ flexGrow: 1 }} />
+            <Box>
+                <LinearProgress sx={{ height: 8 }} />
+            </Box>
+        </Box>
+    );
+};
+
+const ForecastWeather = (props: any) => {
+    const { weatherForecast } = props;
```

Comment:
If we know what this will look like, can we create a type for weatherForecast? Then do `props: { weatherForecast: WeatherForecast }`?